controllers/btpoperator_controller.go

```diff
@@ -498,26 +510,69 @@ func (r *BtpOperatorReconciler) reconcileResources(ctx context.Context, s *corev
 
 	r.deleteCreationTimestamp(resourcesToApply...)
 
+	logger.Info(fmt.Sprintf("check if deployment: %s in %s needs to be restarted.", DeploymentName, resolvedNamespace))
+	clusterIdSecretChanged := r.checkIfOperatorNeedRestart(ctx, resolvedNamespace, baseSecret, &logger)
```

Comment:
`checkIfOperatorNeedRestart` detects if clusterId was changed so the name is misleading.

Reply:
On the other hand the implementation check both cases, so possibly the variable name is inappropriate.